app/marketing/views.py

```diff
@@ -269,6 +269,30 @@ def email_settings(request, key):
     email = ''
     level = ''
     msg = ''
+    email_types = dict([('welcome_mail', 'Welcome Email'),('roundup', 'Round Up'), ('new_bounty_notifications', 'New Bounty Notifications Emails'), ('important_product_updates', 'Product Updates Emails'), ('tip', 'Tip Emails'), ('faucet', 'Faucet Notification Emails'), ('bounty', 'Bounty Notification Emails'), ('bounty_match', 'Bounty Match Emails'), ('bounty_feedback', 'Bounty Feedback Emails'), ('bounty_expiration', 'Bounty Expiration Warning Emails'), ('featured_funded_bounty', 'Featured Funded Bounty Emails')])
+    if request.GET.get('type') in email_types:
+        email = request.user.profile.email
+        if es:
+            key = get_or_save_email_subscriber(email, 'settings')
+            es.email = email
+            unsubscribe_email_type = {}
+            unsubscribe_email_type[request.GET.get('type', '')] = ['1']
```

Comment:
since the if statement on 273 guarantees that that `type` is in the request params, we don't need to use `.get` here, we should just `.get` it once in the function and reuse that instead of calling `.get` multiple times.

Reply:
If I get you correctly, I've created an email_type to get the type in the request params and use that within the if block. Is that what you mean?